### ISSUE TITLE
Drag and Drop to import file in correct account

### ISSUE THREAD

**Issue by sarjanwebdev** [moneymanagerex/moneymanagerex#3684]
# Drop CSV on Account to trigger import
If we allow account name in account list as a drop destination
The import dialog must display to verify settings each time, user can go with preset/default values

# Use cases
We can prefill the file path an account in the import dialog.
This is extension of #3682

**Comment by sarjanwebdev** [comment]
I feel this still holds value

**Comment by n-stein** [comment]
I've enabled file dropping when the _checking_ panel is open to a single account (i.e. not the "All Transactions" view). Dropping a CSV/XML/QIF file opens the corresponding import dialog with the file path, account, and preview data pre-filled.

**Comment by tactilis** [comment]
Tested.  Can't break it 😉

This is actually a _very_ nice improvement to MMEX. I already have the folder open with my .csv files in so drag and drop into MMEX is a lot easier than `Import from | CSV File... / Browse... / find the file... / Open`

I saw from the code that you handle the dropping of multiple files too. Not sure I'd ever do that but nice that it works.

**Comment by n-stein** [comment]
> I saw from the code that you handle the dropping of multiple files too. Not sure I'd ever do that but nice that it works.

We could enable the drop in "All Transactions", the only difference is the account choice wouldn't prefill. Then you could drop a folder with your files for every account in it and it would import them one by one, you just need to set the account (and setting preset currently) each time the dialog opens.

**Comment by tactilis** [comment]
> We could enable the drop in "All Transactions", the only difference is the account choice wouldn't prefill. Then you could drop a folder with your files for every account in it and it would import them one by one, you just need to set the account (and setting preset currently) each time the dialog opens.

That would be OK provided the user has named the files appropriately so they can tell which account each applies to. But they should be doing this in any case even if importing one file at a time so they don't get in a mess.

**Comment by n-stein** [comment]
> If we allow account name in account list as a drop destination

If you drop in the nav panel, should the account be cursor-location dependent (i.e. it will open whatever account your cursor is over when you release the left mouse button)?

**Comment by tactilis** [comment]
> If you drop in the nav panel, should the account be cursor-location dependent (i.e. it will open whatever account your cursor is over when you release the left mouse button)?

I honestly think that it's good enough to make just the transaction list the drop destination. This requires the user to click on the account in the Nav tree and be viewing the account before doing the drag and drop.

I think this is safer because it is then very clear what the context is for the import.

Also, the transaction list is a large target for making the drop, whereas a leaf in the Nav tree is small.

**Comment by n-stein** [comment]
> Also, the transaction list is a large target for making the drop, whereas a leaf in the Nav tree is small.

This is the reason I decided against it on my initial implementation. That and when you are dragging a file into MMEX there is no highlighting to show which node you are hovering over. It is also impossible to add such highlighting because while you are dragging a file from a different application MMEX does not receive mouse events (which I verified by temporarily adding cursor highlighting in the nav panel).

Will leave as-is, I don't think it's a terrible inconvenience to require the target account be clicked first.
